src/main/java/org/osc/manager/ism/api/IsmAgentApi.java

```diff
@@ -16,79 +16,75 @@
  *******************************************************************************/
 package org.osc.manager.ism.api;
 
-import java.text.SimpleDateFormat;
 import java.util.ArrayList;
-import java.util.Date;
 import java.util.List;
-import java.util.concurrent.ThreadLocalRandom;
+
+import javax.persistence.EntityManager;
 
 import org.apache.log4j.Logger;
-import org.osc.manager.ism.model.IsmAgent;
+import org.osc.manager.ism.entities.DeviceMemberEntity;
+import org.osc.manager.ism.entities.DeviceMemberStatusEntity;
 import org.osc.sdk.manager.api.ManagerDeviceMemberApi;
 import org.osc.sdk.manager.element.ApplianceManagerConnectorElement;
 import org.osc.sdk.manager.element.DistributedApplianceInstanceElement;
 import org.osc.sdk.manager.element.ManagerDeviceMemberStatusElement;
 import org.osc.sdk.manager.element.VirtualSystemElement;
-/**
- * Only used for testing
- */
-public final class IsmAgentApi implements ManagerDeviceMemberApi {
+import org.osgi.service.transaction.control.TransactionControl;
 
-    private Logger log = Logger.getLogger(IsmAgentApi.class);
+@SuppressWarnings("unused")
+public final class IsmAgentApi implements ManagerDeviceMemberApi {
+    private static final Logger LOG = Logger.getLogger(IsmAgentApi.class);
+    private IsmDeviceApi api;
+    private final TransactionControl txControl;
```

Comment:
why are these final?

We have to be consistent with declaration. make everything final or leave out final. For osc code, we generally dont declare variables final unless needed.

Reply:
will remove the final.